syntax = "proto3";

package alicepi.vad;

message VadPacket {
  uint64 timestamp_ms = 1;

  oneof payload {
    Audio audio = 2;
    Status status = 3;
  }

  message Audio {
    uint32 sample_rate = 1;
    uint32 channels = 2;
    uint32 sequence = 3;
    bytes data = 4;
  }

  enum Status {
    UNKNOWN = 0;
    SPEECH_DETECTED = 1;
    SPEECH_HANGOVER = 2;
    SILENCE = 3;
  }
}
